### PR TITLE
Academy service crud added

### PR DESCRIPTION
**Pull request by tommygonzaleza**

### CODE REVIEW

File: breathecode/payments/views.py [hunk 1]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+        return handler.response(serializer.data)
+
+    @capable_of('crud_academyservice')
+    def post(self, request, academy_id=None):
+        data = request.data
+        lang = get_user_language(request)
+
+        data['academy'] = academy_id
+
+        if 'price_per_unit' not in request.data:
```
Comment (jefer94) [inline breathecode/payments/views.py:404]: This must be in serializer validation method

File: breathecode/payments/views.py [hunk 2]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+
+    @capable_of('read_academyservice')
+    def get(self, request, academy_id=None):
+        handler = self.extensions(request)
+        lang = get_user_language(request)
+
+        items = AcademyService.objects.filter(academy__id=academy_id)
+        print(items, academy_id)
+
+        if mentorship_service_set_id := request.GET.get('mentorship_service_set_id'):
```
Comment (jefer94) [inline breathecode/payments/views.py:386]: Change to `mentorship_service_set`

File: breathecode/payments/views.py [hunk 3]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+            items = items.filter(mentorship_service_set_id in available_mentorship_service_sets)
+
+        if event_type_set_id := request.GET.get('event_type_set_id'):
```
Comment (jefer94) [inline breathecode/payments/views.py:389]: Change to `event_type_set`

File: breathecode/payments/views.py [hunk 4]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    @capable_of('crud_academyservice')
+    def put(self, request, service_slug=None, academy_id=None):
+        service = Service.objects.filter(Q(owner__id=academy_id) | Q(owner=None), slug=service_slug).first()
+        lang = get_user_language(request)
+
+        if not service:
+            raise ValidationException(translation(lang,
```
Comment (jefer94) [inline breathecode/payments/views.py:424]: This must be in serializer validation method

File: breathecode/payments/views.py [hunk 5]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+                                                  en='Service not found',
+                                                  es='No existe el Servicio',
+                                                  slug='not-found'),
+                                      code=404)
+
+        academyservice = AcademyService.objects.filter(service=service.id, academy__id=academy_id).first()
+
+        if not academyservice:
+            raise ValidationException(translation(lang,
```
Comment (jefer94) [inline breathecode/payments/views.py:433]: This must be in serializer validation method

File: breathecode/payments/views.py [hunk 6]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+                                                  en='Academy Service not found',
+                                                  es='No existe el Academy Service',
+                                                  slug='not-found'),
```
Comment (jefer94) [inline breathecode/payments/views.py:436]: wrong slug

File: breathecode/payments/views.py [hunk 7]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+    @capable_of('crud_academyservice')
+    def put(self, request, service_slug=None, academy_id=None):
+        service = Service.objects.filter(Q(owner__id=academy_id) | Q(owner=None), slug=service_slug).first()
+        lang = get_user_language(request)
+
+        if not service:
+            raise ValidationException(translation(lang,
+                                                  en='Service not found',
+                                                  es='No existe el Servicio',
+                                                  slug='not-found'),
```
Comment (jefer94) [inline breathecode/payments/views.py:427]: wrong slug

File: breathecode/payments/views.py [hunk 8]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+                                      code=404)
```
Comment (jefer94) [inline breathecode/payments/views.py:428]: bad status code

File: breathecode/payments/views.py [hunk 9]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+
+        academyservice = AcademyService.objects.filter(service=service.id, academy__id=academy_id).first()
+
+        if not academyservice:
+            raise ValidationException(translation(lang,
+                                                  en='Academy Service not found',
+                                                  es='No existe el Academy Service',
+                                                  slug='not-found'),
+                                      code=404)
```
Comment (jefer94) [inline breathecode/payments/views.py:437]: bad status code

File: breathecode/payments/views.py [hunk 10]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+        data = request.data
+        lang = get_user_language(request)
+
+        data['academy'] = academy_id
+
+        if 'price_per_unit' not in request.data:
+            raise ValidationException(translation(lang,
+                                                  en='You must specify a price per unit',
+                                                  es='debes especificar un precio por unidad',
+                                                  slug='missing-price_per_unit'),
```
Comment (jefer94) [inline breathecode/payments/views.py:408]: wrong slug

File: breathecode/payments/views.py [hunk 11]
```diff
@@ -371,6 +372,78 @@ def put(self, request, service_slug=None, academy_id=None):
...
+                                      code=404)
```
Comment (jefer94) [inline breathecode/payments/views.py:409]: bad status code

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Comment by tommygonzaleza** [comment]
En el metodo PUT no movi las validaciones para obtener el academyservice porque lo necesito para pasarselo al serializador

**Pull request merged**